realm/realm-library/build.gradle

```diff
@@ -46,6 +46,7 @@ repositories {
 
 dependencies {
     provided 'io.reactivex:rxjava:1.1.0'
+    compile 'com.android.support:recyclerview-v7:23.1.1'
```

Comment:
```
dependencies {
    provided 'io.reactivex:rxjava:1.1.0'
    provided 'com.android.support:recyclerview-v7:23.1.1'
    compile "io.realm:realm-annotations:${version}"

    androidTestCompile 'io.reactivex:rxjava:1.1.0'
    androidTestCompile 'com.android.support:recyclerview-v7:23.1.1'
    androidTestCompile 'com.android.support:support-annotations:23.1.1'
    androidTestCompile 'com.android.support.test:runner:0.4.1'
    androidTestCompile 'com.android.support.test:rules:0.4.1'
    androidTestCompile 'com.google.dexmaker:dexmaker:1.2'
    androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'

    androidTestApt project(':realm-annotations-processor')
}
```
The above dependency list should be what we want, but it outputs the following Gradle error:

```
> Cannot add task ':realm-library:prepareComAndroidSupportRecyclerviewV72311Library' as a task with that name already exists.
```

It looks like a bug in the Android Gradle plugin that doesn't properly handle AAR's that are both `provided` and `androidTestCompile`. The same thing works for JAR's (RxJava). 

Guess we will have to dig a little deeper...